examples/log-mpi.py

```diff
@@ -10,6 +11,24 @@
 from mpi4py import MPI
 
 
+class UserLogQuantity(LogQuantity):
+    """Logging support for arbitrary user quantities."""
+
+    def __init__(self, name, value=None, unit=None,
+          description=None) -> None:
+        LogQuantity.__init__(self, name=name, unit=unit,
+                             description=description)
+        self._quantity_value = value
+
+    def __call__(self) -> float:
+        """Return the actual logged quantity."""
+        return self._quantity_value
+
+    def set_quantity_value(self, value: Any) -> None:
+        """Set the value of the logged quantity."""
+        self._quantity_value = value
```

Comment:
Would it make sense to move this to a place where it's importable?

Reply:
Sorry, somehow this fell off my radar.  I am not accustomed to having logpyle PRs that I'm watching.  

Yes, we could definitely move this and make it "a thing" offered by logpyle - but before we commit to doing that, I just want to point out that this is just an *example* of how to use the capability added by this change set - which was just the part that allows a setter function to be passed into, and then invoked from inside logging.

It would be my preference to add a class similar to this to the `logging_quantities.py` of *MIRGE-Com*, instead of a canned one offered by logpyle, but I defer to @inducer and @matthiasdiener about whether to offer this or similar as a logpyle construct.

Reply:
My vote would be to offer it here... the duplication between the two examples is already a bit much for me to bear. OTOH, if you'd like to demonstrate how (not very) hard it is to make your own `LogQuantity`, then maybe I could be convinced.

Reply:
I see the value in offering a base one.  How about we offer a base one, *and* demonstrate how to make one's own one?

Reply:
:+1:

Reply:
(There could be a comment on the demo one that this specific thing already exists and can just be imported.)